### PR TITLE
Newsletter recipient fixture

### PR DESCRIPTION
**Pull request by NiklasLimberg**
Add NewsletterRecipient Fixture

### CODE REVIEW

File: cypress/support/commands/fixture-commands.js [hunk 1]
```diff
@@ -299,6 +300,20 @@ Cypress.Commands.add('createPromotionFixture', (userData = {}) => {
...
+ * Create newsletter recipient fixture
+ * @memberOf Cypress.Chainable#
+ * @name createNewsletterRecipientFixture
+ * @function
+ */
+Cypress.Commands.add('createNewsletterRecipientFixture', (customer) => {
+    const fixture = new NewsletterRecipientFixture()
+
+    return fixture.setNewsletterRecipientFixture(customer);
+
```
Comment (leichteckig) [inline cypress/support/commands/fixture-commands.js:313]: Leerzeile bitte rausnehmen :)

File: cypress/support/service/administration/fixture/newsletter-recipient.fixture.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                        }
+                    });
+            }
+
+            return getSalutation().then(salutation => {
+                recipient['salutationId'] = salutation.id;
+                return subscribeRecipient(recipient);
+
+            });
+
```
Comment (leichteckig) [inline cypress/support/service/administration/fixture/newsletter-recipient.fixture.js:35]: Hier auch bitte Leerzeichen rausnehmen :)

**Review by leichteckig — CHANGES_REQUESTED** [review]

File: cypress/support/service/administration/fixture/newsletter-recipient.fixture.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                const url = '/sales-channel-api/v3/salutation';
+                return fetch(url, { method: 'GET', headers })
+                    .then((resp) => resp.json())
+                    .then((json) => json.data[0]);
+            }
+
+            function subscribeRecipient(recipient) {
+                const url = '/sales-channel-api/v3/newsletter/subscribe';
+                const body = JSON.stringify(recipient);
+                return fetch(url, { method: 'POST', headers, body })
```
Comment (klarstil) [inline cypress/support/service/administration/fixture/newsletter-recipient.fixture.js:22]: Don't use `fetch` here. `fetch` is not available in Node.js and will be polyfilled with `node-fetch` - see: https://www.npmjs.com/package/node-fetch. Personally I don't see a need to use `fetch` at all. `cy.request()`is what we're using throughout the different kinds of fixture services which worked great for us in the past.

Please refer to the documentation if you have questions how to integrate it: https://docs.cypress.io/api/commands/request.html#Arguments

Additionally you're extending from `AdminFixtureService` which extends from `ApiService` itself. Here we're creating a HTTP client using Axios which can be used as well. Refer to the `ApiService` class regarding the implementation: https://github.com/shopware/e2e-testsuite-platform/blob/master/cypress/support/service/api.service.js

File: cypress/support/service/administration/fixture/newsletter-recipient.fixture.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+const AdminFixtureService = require('../fixture.service.js');
+
+class NewsletterRecipient extends AdminFixtureService {
+    setNewsletterRecipientFixture(recipient) {
+
```
Comment (klarstil) [inline cypress/support/service/administration/fixture/newsletter-recipient.fixture.js:5]: Remove the empty line here pls

**Review by klarstil — CHANGES_REQUESTED** [review]

**Comment by klarstil** [comment]
Additionally to the requested changes in the code., please clean up your commit history. I don't want to see "WIP" commits there. Commits should be meaningful and complete, so in case we need to rollback commits we're not ending up with a broken state.

File: cypress/support/service/administration/fixture/newsletter-recipient.fixture.js (hunk 3 above)
Comment (leichteckig) [reply cypress/support/service/administration/fixture/newsletter-recipient.fixture.js:22]: Agreed, just one little remark here: `cy.request` is not available in the services. Therefore I suggest Promises in our axios context.

**Pull request closed without merging**